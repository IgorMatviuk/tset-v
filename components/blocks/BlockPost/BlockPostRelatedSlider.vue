<template>
  <div class="related-slider">
    <BContainer>
      <div class="related-slider__content">
        <hr class="related-slider__delimiter" />
        <div class="related-slider__heading">
          {{ $t('relatedPosts.title') }}
        </div>
        <div class="related-slider__navigation">
          <SliderArrow arrow-type="prev" />
          <SliderArrow arrow-type="next" />
        </div>
        <div
          v-swiper:postRelatedSlider="swiperOptions"
          :cleanup-styles-on-destroy="false"
          class="related-slider__slider"
        >
          <div class="related-slider__slider-wrapper swiper-wrapper">
            <BlockPostRelatedSliderItem
              v-for="(post, index) in posts"
              :key="index"
              :post="post"
            />
          </div>
          <SliderPagination class="related-slider__pagination" />
        </div>
      </div>
    </BContainer>
  </div>
</template>

<script>
import BlockPostRelatedSliderItem from './BlockPostRelatedSliderItem'
import SliderPagination from '@/components/shared/SliderPagination'
import SliderArrow from '@/components/shared/SliderArrow'

export default {
  components: {
    BlockPostRelatedSliderItem,
    SliderPagination,
    SliderArrow
  },
  props: {
    posts: {
      type: Array,
      required: true
    }
  },
  data() {
    return {
      swiperOptions: {
        slidesPerView: 2,
        spaceBetween: 5,
        watchOverflow: true,
        pagination: {
          el: '.related-slider__pagination',
          clickable: true
        },
        navigation: {
          prevEl: '.related-slider__navigation .swiper-button-prev',
          nextEl: '.related-slider__navigation .swiper-button-next'
        },
        keyboard: {
          enabled: true,
          onlyInViewport: true
        },
        breakpoints: {
          768: {
            slidesPerView: 3,
            spaceBetween: 30
          },
          992: {
            slidesPerView: 4,
            spaceBetween: 30
          }
        }
      }
    }
  }
}
</script>

<style lang="scss">
.related-slider {
  padding-top: 1px;

  &__delimiter {
    margin-top: 8px;
    border-top: 2px solid $primary-light-2;
  }

  &__heading {
    padding-top: 4px;
    padding-bottom: 4px;
    margin-bottom: 10px;
    font-size: 16px;
    line-height: 20px;
    text-transform: uppercase;
    letter-spacing: 0.05em;
  }

  &__slider {
    margin-right: -15px;
    margin-left: -15px;

    @media only screen and (min-width: 540px) {
      margin-right: 0;
      margin-left: 0;
    }
  }

  &__pagination.swiper-pagination {
    position: static;
    margin-top: 8px;
  }

  &__navigation {
    display: none;
  }

  @include media-breakpoint-up(md) {
    margin-bottom: 24px;

    &__content {
      position: relative;
      width: 450px;
      margin-right: auto;
      margin-left: auto;
    }

    &__delimiter {
      margin-bottom: 19px;
    }

    &__heading {
      margin-bottom: 20px;
      font-size: 18px;
      line-height: 23px;
    }

    &__pagination {
      display: none;
    }

    &__navigation {
      position: absolute;
      top: 15px;
      right: 0;
      z-index: 1;
      display: flex;
      justify-content: space-between;
      width: 100px;
      height: 40px;
      margin-right: -5px;
      margin-left: -5px;

      .swiper-button-prev,
      .swiper-button-next {
        position: static;
        margin-top: 0;
      }
    }
  }

  @include media-breakpoint-up(lg) {
    &__content {
      width: 610px;
    }

    &__delimiter {
      margin-top: 6px;
      margin-bottom: 26px;
    }

    &__navigation {
      top: 23px;
    }
  }

  @include media-breakpoint-up(xl) {
    &__content {
      width: 770px;
    }

    &__delimiter {
      margin-top: 0;
    }
  }

  @include media-breakpoint-up(xxl) {
    &__content {
      width: 930px;
    }

    &__delimiter {
      margin-top: 14px;
      margin-bottom: 37px;
    }

    &__navigation {
      top: 31px;
    }
  }
}
</style>
