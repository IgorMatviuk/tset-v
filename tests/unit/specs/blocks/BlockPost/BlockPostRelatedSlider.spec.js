import { mount, createLocalVue } from '@vue/test-utils'
import _cloneDeep from 'lodash.clonedeep'

import { BContainer } from 'bootstrap-vue'
import VueAwesomeSwiper from 'vue-awesome-swiper'

import BlockPostRelatedSlider from '@/components/blocks/BlockPost/BlockPostRelatedSlider'

import createFactory from '@/api/helpers/factory'
import postAttrs from '@/api/factories/post'

const localVue = createLocalVue()
localVue.use(VueAwesomeSwiper)
localVue.component('BContainer', BContainer)

describe('BlockPostRelatedSlider', () => {
  let posts

  function createBlockPostRelatedSlider() {
    const wrapper = mount(BlockPostRelatedSlider, {
      localVue,
      propsData: {
        posts: _cloneDeep(posts)
      },
      mocks: {
        $i18n: {
          path: (url) => url
        },
        $t: (path) => path
      },
      stubs: ['NuxtLink', 'svg-icon']
    })

    return wrapper
  }

  beforeEach(() => {
    posts = createFactory(postAttrs, 5)
  })

  test('render related posts slider', () => {
    const wrapper = createBlockPostRelatedSlider()

    expect(wrapper.element).toMatchSnapshot()
  })
})
