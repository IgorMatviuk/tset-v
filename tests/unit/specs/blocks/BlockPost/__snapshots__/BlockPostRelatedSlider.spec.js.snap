// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockPostRelatedSlider render related posts slider 1`] = `
<div
  class="related-slider"
>
  <div
    class="container"
  >
    <div
      class="related-slider__content"
    >
      <hr
        class="related-slider__delimiter"
      />
       
      <div
        class="related-slider__heading"
      >
        
        relatedPosts.title
      
      </div>
       
      <div
        class="related-slider__navigation"
      >
        <div
          class="swiper-button-prev"
        >
          <svg-icon-stub
            class="swiper-button-prev-icon"
            height="9"
            name="arrow"
            width="30"
          />
        </div>
         
        <div
          class="swiper-button-next"
        >
          <svg-icon-stub
            class="swiper-button-next-icon"
            height="9"
            name="arrow"
            width="30"
          />
        </div>
      </div>
       
      <div
        class="related-slider__slider swiper-container swiper-container-initialized swiper-container-horizontal"
      >
        <div
          class="related-slider__slider-wrapper swiper-wrapper"
          style="webkit-transition-duration: 0ms; transition-duration: 0ms; webkit-transform: translate3d(NaNpx, 0px, 0px); transform: translate3d(NaNpx, 0px, 0px);"
        >
          <div
            class="related-slider__slide swiper-slide"
          >
            <div
              class="related-post"
            >
              <div
                class="related-post__preview"
              >
                <div
                  class="related-post__image-wrapper"
                >
                  <div
                    class="img-loader"
                  >
                    <picture>
                      <source
                        data-srcset="/img/posts/post_xs.webp"
                        srcset="/img/stub.png"
                        type="image/webp"
                      />
                       
                      <img
                        alt="Совет в декрет! Как подобрать стильные летние платья и сарафаны для беременных?"
                        class="related-post__image lazyload"
                        data-object-fit="cover"
                        data-object-position="center"
                        data-src="/img/posts/post_xs.jpg"
                        loading="lazy"
                        src="/img/stub.png"
                        title="Совет в декрет! Как подобрать стильные летние платья и сарафаны для беременных?"
                      />
                    </picture>
                  </div>
                </div>
              </div>
               
              <nuxtlink-stub
                class="related-post__title"
                to="blog/sovet-v-dekret-kak-podobrat-stilnye-letnie-platya-i-sarafany-dlya-beremennyh"
              >
                Совет в декрет! Как подобрать стильные летние платья и сарафаны для беременных?
              </nuxtlink-stub>
               
              <div
                class="related-post__date"
              >
                
    17 июля 2018
  
              </div>
            </div>
          </div>
          <div
            class="related-slider__slide swiper-slide"
          >
            <div
              class="related-post"
            >
              <div
                class="related-post__preview"
              >
                <div
                  class="related-post__image-wrapper"
                >
                  <div
                    class="img-loader"
                  >
                    <picture>
                      <source
                        data-srcset="/img/posts/post_xs.webp"
                        srcset="/img/stub.png"
                        type="image/webp"
                      />
                       
                      <img
                        alt="Совет в декрет! Как подобрать стильные летние платья и сарафаны для беременных?"
                        class="related-post__image lazyload"
                        data-object-fit="cover"
                        data-object-position="center"
                        data-src="/img/posts/post_xs.jpg"
                        loading="lazy"
                        src="/img/stub.png"
                        title="Совет в декрет! Как подобрать стильные летние платья и сарафаны для беременных?"
                      />
                    </picture>
                  </div>
                </div>
              </div>
               
              <nuxtlink-stub
                class="related-post__title"
                to="blog/sovet-v-dekret-kak-podobrat-stilnye-letnie-platya-i-sarafany-dlya-beremennyh"
              >
                Совет в декрет! Как подобрать стильные летние платья и сарафаны для беременных?
              </nuxtlink-stub>
               
              <div
                class="related-post__date"
              >
                
    17 июля 2018
  
              </div>
            </div>
          </div>
          <div
            class="related-slider__slide swiper-slide"
          >
            <div
              class="related-post"
            >
              <div
                class="related-post__preview"
              >
                <div
                  class="related-post__image-wrapper"
                >
                  <div
                    class="img-loader"
                  >
                    <picture>
                      <source
                        data-srcset="/img/posts/post_xs.webp"
                        srcset="/img/stub.png"
                        type="image/webp"
                      />
                       
                      <img
                        alt="Совет в декрет! Как подобрать стильные летние платья и сарафаны для беременных?"
                        class="related-post__image lazyload"
                        data-object-fit="cover"
                        data-object-position="center"
                        data-src="/img/posts/post_xs.jpg"
                        loading="lazy"
                        src="/img/stub.png"
                        title="Совет в декрет! Как подобрать стильные летние платья и сарафаны для беременных?"
                      />
                    </picture>
                  </div>
                </div>
              </div>
               
              <nuxtlink-stub
                class="related-post__title"
                to="blog/sovet-v-dekret-kak-podobrat-stilnye-letnie-platya-i-sarafany-dlya-beremennyh"
              >
                Совет в декрет! Как подобрать стильные летние платья и сарафаны для беременных?
              </nuxtlink-stub>
               
              <div
                class="related-post__date"
              >
                
    17 июля 2018
  
              </div>
            </div>
          </div>
          <div
            class="related-slider__slide swiper-slide"
          >
            <div
              class="related-post"
            >
              <div
                class="related-post__preview"
              >
                <div
                  class="related-post__image-wrapper"
                >
                  <div
                    class="img-loader"
                  >
                    <picture>
                      <source
                        data-srcset="/img/posts/post_xs.webp"
                        srcset="/img/stub.png"
                        type="image/webp"
                      />
                       
                      <img
                        alt="Совет в декрет! Как подобрать стильные летние платья и сарафаны для беременных?"
                        class="related-post__image lazyload"
                        data-object-fit="cover"
                        data-object-position="center"
                        data-src="/img/posts/post_xs.jpg"
                        loading="lazy"
                        src="/img/stub.png"
                        title="Совет в декрет! Как подобрать стильные летние платья и сарафаны для беременных?"
                      />
                    </picture>
                  </div>
                </div>
              </div>
               
              <nuxtlink-stub
                class="related-post__title"
                to="blog/sovet-v-dekret-kak-podobrat-stilnye-letnie-platya-i-sarafany-dlya-beremennyh"
              >
                Совет в декрет! Как подобрать стильные летние платья и сарафаны для беременных?
              </nuxtlink-stub>
               
              <div
                class="related-post__date"
              >
                
    17 июля 2018
  
              </div>
            </div>
          </div>
          <div
            class="related-slider__slide swiper-slide"
          >
            <div
              class="related-post"
            >
              <div
                class="related-post__preview"
              >
                <div
                  class="related-post__image-wrapper"
                >
                  <div
                    class="img-loader"
                  >
                    <picture>
                      <source
                        data-srcset="/img/posts/post_xs.webp"
                        srcset="/img/stub.png"
                        type="image/webp"
                      />
                       
                      <img
                        alt="Совет в декрет! Как подобрать стильные летние платья и сарафаны для беременных?"
                        class="related-post__image lazyload"
                        data-object-fit="cover"
                        data-object-position="center"
                        data-src="/img/posts/post_xs.jpg"
                        loading="lazy"
                        src="/img/stub.png"
                        title="Совет в декрет! Как подобрать стильные летние платья и сарафаны для беременных?"
                      />
                    </picture>
                  </div>
                </div>
              </div>
               
              <nuxtlink-stub
                class="related-post__title"
                to="blog/sovet-v-dekret-kak-podobrat-stilnye-letnie-platya-i-sarafany-dlya-beremennyh"
              >
                Совет в декрет! Как подобрать стильные летние платья и сарафаны для беременных?
              </nuxtlink-stub>
               
              <div
                class="related-post__date"
              >
                
    17 июля 2018
  
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="swiper-pagination related-slider__pagination"
        />
        <span
          aria-atomic="true"
          aria-live="assertive"
          class="swiper-notification"
        />
      </div>
    </div>
  </div>
</div>
`;
